doc/ingest.md

```diff
@@ -86,6 +88,35 @@ The network indexer nodes expect that metadata begins with a `uvarint` identifyi
   * the proposed `uvarint` protocol is `0x3D0000`.
   * the following bytes are not yet defined.
 
+#### ExtendedProvider
+
+The `ExtendedProvider` field allows for specification of provider families, in cases where a provider operates multiple PeerIDs, perhaps with different transport protocols between them, but over the same database of content.
+
+```
+type ExtendedProvider struct {
+    Providers [Provider]
+    Override bool
+}
+
+type Provider struct {
+    ID String
+    Addresses [String]
+    Metadata optional Bytes
+    Signature Bytes
+}
+```
+
+* If `Metadata` is not specified for a `Provider`, the metadata from the encapsulating  `Advertisement` will be used instead.
+* If a `Provider` listing is written with no `ContextID`, those peers will be returned for all advertisements published by the publisher.
+  * If `Override` is set on an `ExtendedProvider` entry on an advertisement with a `ContextID`, it indicates that any specified chain-level set of providers should not be returned for that context ID. `Providers` will be returned Instead.
+  * If `Override` is not set on an entry for an advertisement with a `ContextID`, it will be combined as a union with any set chain-level `ExtndedProvider`s.
+  * If `Override` is set on `ExtendedProvider` for an advertisement without a `ContextID`, the entry is invalid and should be ignored.
```

Comment:
What about the case of a removal advertisement (`IsRm == true`) that does not have a context ID?  Would that be valid and indicate that a set of providers is being removed, or should that be done with separate removal advertisements?